modules/client/src/schemas.ts

```diff
@@ -0,0 +1,66 @@
+import { array, object, string } from "yup";
+import {
+  addressOrEnsSchema,
+  addressSchema,
+  bigintSchema,
+  ipfsUriSchema,
+  permissionSchema,
+  pluginInstallItemSchema,
+  uint8ArraySchema,
+} from "./client-common";
+
+// DAO details
+const daoResourceLinkSchema = object({
+  name: string().required(),
+  url: string().required(),
+});
+
+export const metadataSchema = object({
+  name: string().required(),
+  description: string().required(),
+  avatar: string().optional(),
+  links: array(daoResourceLinkSchema).required(),
+});
+// Create proposal
+export const createParamsSchema = object({
+  metadataUri: ipfsUriSchema.required(),
+  ensSubdomain: string().required(),
+  trustedForwarder: string().optional(),
+  plugins: array(pluginInstallItemSchema).required(),
+});
+
+// Withdraw
+export const withdrawParamsSchema = object({
+  recipientAddress: addressOrEnsSchema.required(),
+  amount: bigintSchema.required(),
+  tokenAddress: addressSchema,
+  reference: string(),
+});
+
+// deposit
+export const depositParamsSchema = object({
+  daoAddressOrEns: addressOrEnsSchema.required(),
+  amount: bigintSchema.required(),
+  tokenAddress: addressSchema,
+  reference: string(),
+});
+
+// has permission
+export const hasPermissionParamsSchema = object({
+  daoAddressOrEns: addressOrEnsSchema.required(),
+  where: addressOrEnsSchema.required(),
+  who: addressOrEnsSchema.required(),
+  permission: permissionSchema.required(),
```

Comment:
I'm not sure if we call it `permissions ` everywhere here. In general we speak of permission ID's when computed in Solidity, but the variable names are just permission. 
Feel free to resolve

Reply:
Validate string regex `A-Z_`

Reply:
Added validation function `isPermission` that solves this